// @puzzlehq/types@1.0.8/index.ts
/// types

import { z } from 'zod';

export type Account = {
  _id: string;
  address: string;
  viewkey: string;
  lastSyncedHeight: number;
};

export type Event = {
  _id: string;
  type: EventType;
  owner: string;

  status: EventStatus;
  created: Date;
  broadcast?: Date;
  broadcastHeight?: number;
  settled?: Date;

  network: Network;
  transactionId?: string;
  transitionId?: string;
  height?: number;
  description?: string;
  fee: number;
  programId: string;
  functionId: string;
  visibility: Visibility;
  inputs: (string | null)[];
  outputs: (string | null)[];
  error?: string;
};

export type EventCreate = {
  type: string;
  owner: string;
  network: string;
  functionId?: string;
  programId?: string;
  visibility?: string;
  inputs?: string[];
  outputs?: string[];
};

export type Record = {
  _id: string;
  eventId: string;
  height: number;
  ciphertext: string;
  programId: string;
  functionId: string;
  transitionId: string;
  transactionId: string;
  owner: string;
  spent: boolean;
  serialNumber?: string;
};

export type Asset = {
  type: AssetType;
};

export type AssetValue = {
  asset: Asset;
  value: BigInt;
};

/// enums

export enum EventType {
  Deploy = 'Deploy',
  Execute = 'Execute',
  Send = 'Send',
  Receive = 'Receive',
  Join = 'Join',
  Split = 'Split',
  Shield = 'Shield',
  Unshield = 'Unshield',
}

export enum EventStatus {
  Creating = 'Creating',
  Pending = 'Pending',
  Settled = 'Settled',
  Failed = 'Failed',
}

export enum Visibility {
  Private = 'Private',
  Public = 'Public',
}

export enum Network {
  AleoTestnet = 'AleoTestnet',
  AleoMainnet = 'AleoMainnet',
}

export enum AssetType {
  ALEO,
}

export const zodEventType = z.nativeEnum(EventType);
export const zodEventStatus = z.nativeEnum(EventStatus);
export const zodNetwork = z.nativeEnum(Network);
export const zodVisibility = z.nativeEnum(Visibility);

export * from './aleo.js';
