// @puzzlehq/types@1.0.8/index.js
/// types
import { z } from 'zod';
/// enums
export var EventType;
(function (EventType) {
    EventType["Deploy"] = "Deploy";
    EventType["Execute"] = "Execute";
    EventType["Send"] = "Send";
    EventType["Receive"] = "Receive";
    EventType["Join"] = "Join";
    EventType["Split"] = "Split";
    EventType["Shield"] = "Shield";
    EventType["Unshield"] = "Unshield";
})(EventType || (EventType = {}));
export var EventStatus;
(function (EventStatus) {
    EventStatus["Creating"] = "Creating";
    EventStatus["Pending"] = "Pending";
    EventStatus["Settled"] = "Settled";
    EventStatus["Failed"] = "Failed";
})(EventStatus || (EventStatus = {}));
export var Visibility;
(function (Visibility) {
    Visibility["Private"] = "Private";
    Visibility["Public"] = "Public";
})(Visibility || (Visibility = {}));
export var Network;
(function (Network) {
    Network["AleoTestnet"] = "AleoTestnet";
    Network["AleoMainnet"] = "AleoMainnet";
})(Network || (Network = {}));
export var AssetType;
(function (AssetType) {
    AssetType[AssetType["ALEO"] = 0] = "ALEO";
})(AssetType || (AssetType = {}));
export const zodEventType = z.nativeEnum(EventType);
export const zodEventStatus = z.nativeEnum(EventStatus);
export const zodNetwork = z.nativeEnum(Network);
export const zodVisibility = z.nativeEnum(Visibility);
export * from './aleo.js';
